verification/src/solidity/compiler_fetcher.rs

```diff
@@ -0,0 +1,295 @@
+use crate::compiler::{
+    fetcher::Fetcher,
+    version::{CompilerVersion, ParseError},
+};
+use async_trait::async_trait;
+use serde::Deserialize;
+use std::{
+    collections::HashMap,
+    fs::{File, OpenOptions},
+    io::ErrorKind,
+    os::unix::prelude::OpenOptionsExt,
+    path::{Path, PathBuf},
+    str::FromStr,
+};
+use thiserror::Error;
+use url::Url;
+
+#[derive(Error, Debug)]
+pub enum ListError {
+    #[error("fetching list json returned error: {0}")]
+    ListJsonFetch(reqwest::Error),
+    #[error("cannot parse list json file: {0}")]
+    ParseListJson(reqwest::Error),
+    #[error("error parsing tag name into version: {0}")]
+    TagName(ParseError),
+    #[error("error parsing 'path' field: {0}")]
+    Path(url::ParseError),
+}
+
+#[derive(Debug, Deserialize, PartialEq)]
+pub struct ListJson {
+    pub builds: Vec<CompilerInfo>,
+}
```

Comment:
Maybe keep builds just inside a `CompilerFetcher`? Why do we need a separate structure? We actually require that information even when downloading compilers (e.g. not implemented now but are going to add validating checsum on download)